src/librustc_resolve/diagnostics.rs

```diff
@@ -93,6 +93,41 @@ mod bar {
 ```
 "##,
 
+E0254: r##"
+This error indicates that an imported symbol resolved to an already imported
+symbol.
+
+An example of this error:
+
+```
+// crate foo
+pub use foo::bar;
+pub mod foo; // note that there's a `pub` here
+```
+
+```
+// other program
+extern crate foo;
+
+use foo::*; // error, `foo::*` tries to bind `foo::foo` to `foo`
```

Comment:
Okay so here is what I think is a more direct example:

This works for any external crate, but I'll include a trivial one that provides zero exports.  (But if you want to add a one line definition to it, that's fine, as long as its not named `foo` too.)

external crate foo.rs:
```rust
#![crate_type="lib"]
```

client code bar.rs:
```
extern crate foo; // this puts `foo` into the root type+module namespace

use inner_mod::foo; // this also tries to put `foo` in the same namespace

mod inner_mod {
    pub mod foo { }
}
```

And then one might note that this variant of `bar.rs` also has the same problem:

```rust
extern crate foo; // this puts `foo` into the root type+module namespace

use inner_mod::foo; // this also tries to put `foo` in the same namespace

mod inner_mod {
    pub type foo = i32;
}
```